cylc/uiserver/scripts/gui.py

```diff
@@ -53,25 +54,78 @@ def main(*argv):
     jp_server_opts, new_gui, workflow_id = parse_args_opts()
     if '--help' not in sys.argv:
         # get existing jpserver-<pid>-open.html files
-        # assume if this exists then the server is still running
-        # these files are cleaned by jpserver on shutdown
+        # check if the server is available for use
+        # prompt for user whether to clean files for un-usable uiservers
+        # these files are usually cleaned by jpserver on shutdown, although
+        # can be left behind on crash or a `kill -9` of the process
         existing_guis = glob(os.path.join(INFO_FILES_DIR, "*open.html"))
         if existing_guis and not new_gui:
-            gui_file = random.choice(existing_guis)
-            print(
-                "Opening with existing gui." +
-                f" Use {CLI_OPT_NEW} option for a new gui.",
-                file=sys.stderr
-            )
-            update_html_file(gui_file, workflow_id)
-            if '--no-browser' not in sys.argv:
-                webbrowser.open(f'file://{gui_file}', autoraise=True)
-            return
+            url = select_info_file(existing_guis)
+            if url:
+                print(
+                    "Opening with existing gui." +
+                    f" Use {CLI_OPT_NEW} option for a new gui.",
+                    file=sys.stderr
+                )
+                url = update_url(url, workflow_id)
+                if '--no-browser' not in sys.argv:
+                    webbrowser.open(url, autoraise=True)
+                return
     return CylcUIServer.launch_instance(
         jp_server_opts or None, workflow_id=workflow_id
     )
 
 
+def select_info_file(existing_guis: list) -> Optional[str]:
+    """This will select an active ui-server info file"""
+    existing_guis.sort(key=os.path.getmtime, reverse=True)
+    for gui_file in existing_guis:
+        url = fish_url_from_file(gui_file)
+        if url and is_active_gui(url):
+            return url
+        check_remove_file(gui_file)
+    return None
+
+
+def fish_url_from_file(gui_file):
+    with open(gui_file, "r") as f:
+        file_content = f.read()
+    url_extract_regex = re.compile('url=(.*?)\"')
+    match = url_extract_regex.search(file_content)
+    return match.group(1) if match else None
+
+
+def is_active_gui(url):
+    """Returns true if return code is 200 from server"""
+    try:
+        req = requests.get(url)
+        if req.status_code == 200:
+            return True
+    except RequestException:
+        return False
+
+
+def clean_info_files(gui_file):
+    pid = re.compile(r'-(\d*)-open\.html').search(gui_file).group(1)
+    json_file = os.path.join(INFO_FILES_DIR, f"jpserver-{pid}.json")
+    try:
+        os.unlink(gui_file)
+        os.unlink(json_file)
+    except Exception:
+        pass
+
+
+def check_remove_file(gui_file) -> None:
+    """Ask user if they want to remove the file."""
+    print("The following file cannot be used to open the Cylc GUI:"
```

Comment:
IMO this should be amber to draw attention to the fact that the program wants input in the terminal - this isn't very common for the UIS.